### PR TITLE
nagios_xi: fix handling of potentially null method call receivers

### PR DESCRIPTION
**Pull request by efd6**
## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ] 

## Related issues

- For #8646

### CODE REVIEW

**Review by ShourieG — APPROVED** [review]
LGTM

File: packages/nagios_xi/data_stream/service/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -681,7 +703,7 @@ processors:
 
 - remove:
     field: event.original
-    if: "ctx?.tags == null || !(ctx.tags.contains('preserve_original_event'))"
+    if: "ctx.tags == null || !(ctx.tags.contains('preserve_original_event'))"
```
Comment (muthu-mps) [inline packages/nagios_xi/data_stream/service/elasticsearch/ingest_pipeline/default.yml:706]: Can we also make this change in the `event` datastream? This way the change becomes consistent across all datastreams.

Comment (efd6) [reply packages/nagios_xi/data_stream/service/elasticsearch/ingest_pipeline/default.yml:706]: Stretching scope, but done.

**Review by muthu-mps — APPROVED** [review]
LGTM!

**Pull request merged**